### PR TITLE
feat(dashboard): Display issue widgets in dashboard View Mode

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Frontend changes to display Issue widgets on the dashboard when in view mode.
Requires backend issue changes to properly detect widget type in order to render issue widgets (merge with this branch to test https://github.com/getsentry/sentry/pull/30090)
image

### CODE REVIEW

**Review by narsaynorath — COMMENTED** [review]
Just some small FE comments for now!

**Comment by edwardgou-sentry** [comment]
Fixed accidental backend branch merge into pr

File: tests/js/spec/views/dashboardsV2/sortableWidget.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        isEditing={false}
+        onDelete={() => undefined}
+        onEdit={() => undefined}
+      />,
+      initialData.routerContext
+    );
+    expect(wrapper.find('IssueWidgetCard').length).toBe(0);
+    expect(wrapper.find('WidgetCard').length).toBe(1);
+  });
+  it('renders an IssueWidgetCard', async () => {
```
Comment (narsaynorath) [inline tests/js/spec/views/dashboardsV2/sortableWidget.spec.tsx:45]: This test asserts `expect(wrapper.find('IssueWidgetCard').length).toBe(0);`, I think you intended to update the `widgetType` to be for an Issues widget and assert the length is 1?

File: static/app/views/dashboardsV2/dashboard.tsx [hunk 2]
```diff
@@ -166,15 +169,24 @@ class Dashboard extends Component<Props> {
...
-    openAddDashboardWidgetModal({
-      organization,
-      dashboard,
-      widget,
-      selection,
-      onAddWidget: this.handleAddComplete,
-      onUpdateWidget: this.handleUpdateComplete(index),
-    });
+    if (widget.widgetType === WidgetType.ISSUE) {
+      openAddDashboardIssueWidgetModal({
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/views/dashboardsV2/dashboard.tsx]: 👍  Agreed on this, cleaned it up!

File: static/app/views/dashboardsV2/issueWidgetCard.tsx [hunk 3]
```diff
@@ -117,6 +119,44 @@ class IssueWidgetCard extends React.Component<Props> {
...
+          <IconClick>
+            <StyledIconGrabbable
+              color="textColor"
+              {...draggableProps?.listeners}
+              {...draggableProps?.attributes}
+            />
+          </IconClick>
+          <IconClick
+            data-test-id="widget-edit"
+            onClick={() => {
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/views/dashboardsV2/issueWidgetCard.tsx]: Good call, updated!

File: tests/js/spec/views/dashboardsV2/sortableWidget.spec.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply tests/js/spec/views/dashboardsV2/sortableWidget.spec.tsx:45]: 😅  good catch, forgot to update this! Done!

**Review by narsaynorath — APPROVED** [review]
+1,  looks good! I'm able to add widgets to my dashboard in view mode

**Pull request merged**